timemachine/potentials/nonbonded.py

```diff
@@ -565,31 +648,30 @@ def lj_prefactors_on_traj(traj, boxes, sigmas, epsilons, ligand_indices, env_ind
     """
     validate_interaction_group_idxs(len(traj[0]), ligand_indices, env_indices)
 
-    sig_ligand = sigmas[ligand_indices]
-
     eps_env = epsilons[env_indices]
     sig_env = sigmas[env_indices]
 
     def f_snapshot(coords, box):
         x_ligand = coords[ligand_indices]
         x_env = coords[env_indices]
-        return lj_prefactors_on_snapshot(x_ligand, x_env, sig_ligand, sig_env, eps_env, box, cutoff)
+        return lj_prefactors_on_snapshot(x_ligand, x_env, sig_env, eps_env, box, cutoff)
 
     return vmap(f_snapshot)(traj, boxes)
```

Comment:
@maxentile This is the part that can use a lot of memory. The intermediate computation of `r_env = jax_utils.distance_from_one_to_others(...)` is where most of the memory is allocated. I handled this by breaking up the traj into smaller segments, it could be useful to add this as an option to `lj_prefactors_on_traj`. Same with `coulomb_prefactors_on_traj`.

Reply:
256 should be good (I've been using 200)

Reply:
Reduced to 200 to match https://github.com/proteneer/timemachine/pull/931/commits/27d0dc9711f7025980c4d907712e795b16b7bf9f

Reply:
I might be missing something but it's not clear to me why `jax_utils.distance_from_one_to_others` should be allocating a lot of memory -- seems like it should just be linear in `x_others`?

Reply:
I think the size of this intermediate array may be something like `traj_length * n_ligand_atoms * n_environment_atoms * 3` (but I'm not sure). Although linear in `traj_length`, the constant factor is probably large.

Follow-ups could be:
* to provide Jax some clearer hint to "fuse" the computation distance matrix with the sum reduction / avoid this large intermediate array
* to write a "custom op" implementation that uses the neighborlist

I'm leaning towards just doing the second option, since
(1) there would also be utility in computing these on the fly (to avoid ever having to store trajectories),
(2) I don't foresee a need to use any other jax transformations with these preprocessing functions (I don't see a compelling reason to keep the implementation in Jax if doing so is inconvenient / introduces surprising resource requirements)